### PR TITLE
Rest ingest

### PR DESCRIPTION
**Pull request by jnm92**
Add rest enabled endpoint for ingest command + add integration tests for ingest and file upload

### CODE REVIEW

File: services/rest/src/test/java/mil/nga/giat/geowave/service/rest/RestServerTest.java [hunk 1]
```diff
@@ -440,4 +443,200 @@ public void callWithMissingParameter()
 
 		fail("No exception was thrown");
 	}
+
+	// Test ingesting data into a newly created store with spatial index
+	@Test
+	public void geowave_ingest()
```
Comment (mak438) [inline services/rest/src/test/java/mil/nga/giat/geowave/service/rest/RestServerTest.java:519]: This test is probably redundant given the other one.

File: services/rest/src/main/java/mil/nga/giat/geowave/service/rest/FileUpload.java [hunk 2]
```diff
@@ -106,6 +106,11 @@ public UploadedFile accept(
 			// 3/ Request is parsed by the handler which generates a
 			// list of FileItems
 			String tempDir = System.getProperty("java.io.tmpdir");
+			// Linux returns temp dir as '/tmp' but need to add '/' after to
+			// separate file name
+			if (tempDir.charAt(tempDir.length() - 1) != '/' && tempDir.charAt(tempDir.length() - 1) != '\\') {
+				tempDir = tempDir + '/';
+			}
```
Comment (xm-gui) [inline services/rest/src/main/java/mil/nga/giat/geowave/service/rest/FileUpload.java]: Haven't tested this yet, but maybe we could change line 120 to: 
result = new UploadedFile(filename.getAbsolutePath());

Comment (jnm92) [reply services/rest/src/main/java/mil/nga/giat/geowave/service/rest/FileUpload.java]: Good suggestion, thanks

**Review by mak438 — APPROVED** [review]
Looks good.

**Pull request merged**